python/catkin/builder.py

```diff
@@ -287,17 +288,8 @@ def get_multiarch():
 
 def get_python_install_dir():
     # this function returns the same value as the CMake variable PYTHON_INSTALL_DIR from catkin/cmake/python.cmake
```

Comment:
The build tool is also using `get_python_install_dir` in order to do additional stuff, e.g. generating additional environment variables. So the constraint that the Python versions needs to match is still the case. But I don't think that this is affected in any way by this patch. It was the case before as well as after this change.